docs/hooks.rst

```diff
@@ -0,0 +1,105 @@
+.. _hooks:
+
+Hook Scripts
+============
+
+Similar to any other testing framework, Dredd supports executing code around each test step. Hooks are code blocks executed in defined stage of :ref:`execution lifecycle <execution-life-cycle>`. In the hooks code you have an access to compiled HTTP :ref:`transaction object <transaction-object-structure>` which you can modify.
+
+Hooks are usually used for:
+
+-  loading db fixtures
+-  cleanup after test step or steps
+-  handling authentication and sessions
+-  passing data between transactions (saving state from responses to *stash*)
+-  modifying request generated from API description
+-  changing generated expectations
+-  setting custom expectations
+-  debugging via logging stuff
+
+Languages
+---------
+
+You can interact with your server implementation in following languages:
+
+-  :ref:`Go <hooks-go>`
```

Comment:
Perhaps worth moving the hooks into a directory, for example:

```
hooks/
  index.rst
  go.rst
  nodejs.rst
```

etc, then you can make use of toctree to automatically produce this list.

Something like:

```
.. toctree:
  .
```

The only problem, is that this can be a breaking change for the URLs, but perhaps worth it using redirect, will leave to your judgement though (and perhaps should be changed separately if  it is).

Reply:
That's a good idea 👍 I'll play with that as a follow-up.